bindings/wasm/helpers.cpp

```diff
@@ -0,0 +1,222 @@
+  }
+  if (mesh["halfedgeTangent"] != val::undefined()) {
+    out.halfedgeTangent =
+        convertJSArrayToNumberVector<float>(mesh["halfedgeTangent"]);
+  }
+  if (mesh["runTransform"] != val::undefined()) {
+    out.runTransform =
+        convertJSArrayToNumberVector<float>(mesh["runTransform"]);
+  }
+  return out;
+}
+
+val GetMeshJS(const Manifold& manifold, const glm::ivec3& normalIdx) {
+  MeshGL mesh = manifold.GetMeshGL(normalIdx);
+  return MeshGL2JS(mesh);
+}
+
+val Merge(const val& mesh) {
+  val out = val::object();
+  MeshGL meshGL = MeshJS2GL(mesh);
+  bool changed = meshGL.Merge();
+  out.set("changed", changed);
+  out.set("mesh", changed ? MeshGL2JS(meshGL) : mesh);
+  return out;
+}
+
+Manifold Smooth(const val& mesh,
+                const std::vector<Smoothness>& sharpenedEdges = {}) {
+  return Manifold::Smooth(MeshJS2GL(mesh), sharpenedEdges);
+}
+
+}  // namespace js
+
+namespace cross_js {
+CrossSection OfPolygons(std::vector<std::vector<glm::vec2>> polygons,
+                        int fill_rule) {
+  auto fr = fill_rule == 0   ? CrossSection::FillRule::EvenOdd
+            : fill_rule == 1 ? CrossSection::FillRule::NonZero
+            : fill_rule == 2 ? CrossSection::FillRule::Positive
+                             : CrossSection::FillRule::Negative;
+  return CrossSection(polygons, fr);
+}
+
+CrossSection Union(const CrossSection& a, const CrossSection& b) {
+  return a + b;
+}
+
+CrossSection Difference(const CrossSection& a, const CrossSection& b) {
+  return a - b;
+}
+
+CrossSection Intersection(const CrossSection& a, const CrossSection& b) {
+  return a ^ b;
+}
+
+CrossSection UnionN(const std::vector<CrossSection>& cross_sections) {
+  return CrossSection::BatchBoolean(cross_sections, OpType::Add);
+}
+
+CrossSection DifferenceN(const std::vector<CrossSection>& cross_sections) {
+  return CrossSection::BatchBoolean(cross_sections, OpType::Subtract);
+}
+
+CrossSection IntersectionN(const std::vector<CrossSection>& cross_sections) {
+  return CrossSection::BatchBoolean(cross_sections, OpType::Intersect);
+}
+
+CrossSection Transform(CrossSection& cross_section, const val& mat) {
+  std::vector<float> array = convertJSArrayToNumberVector<float>(mat);
+  glm::mat3x2 matrix;
+  for (const int col : {0, 1, 2})
+    for (const int row : {0, 1}) matrix[col][row] = array[col * 3 + row];
+  return cross_section.Transform(matrix);
+}
+
+CrossSection Warp(CrossSection& cross_section, uintptr_t funcPtr) {
+  void (*f)(glm::vec2&) = reinterpret_cast<void (*)(glm::vec2&)>(funcPtr);
+  return cross_section.Warp(f);
+}
+
+CrossSection Offset(CrossSection& cross_section, double delta, int join_type,
+                    double miter_limit, double arc_tolerance) {
+  auto jt = join_type == 0   ? CrossSection::JoinType::Square
+            : join_type == 1 ? CrossSection::JoinType::Round
+                             : CrossSection::JoinType::Miter;
+  return cross_section.Offset(delta, jt, miter_limit, arc_tolerance);
```

Comment:
Let's do this in a separate PR, but I agree with @pca006132 that instead of `arc_tolerance` we should use our `setMinCircularAngle` etc. functions to define this.